feat: Optionally hide mode switch in Calendar

BenjaVR opened the pull request:
<!--
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]
-->
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

When you use the Calendar component, there will always be a switch to change the mode from "month" to "year" and vice versa. In some use cases you might want to show the Calendar only in on of the two modes, without the option to switch.

### 💡 Solution

This can be solved by passing a prop (`showModeSwitch`) to the Calendar (and Header) component, to optionally hide the mode switch.

### 📝 Changelog

<!--
Describe changes from userside, and list all potential break changes or other risks.
--->

- English Changelog:
- Chinese Changelog (optional):

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed


-----
[View rendered components/calendar/demo/no-mode-switch.md](https://github.com/BenjaVR/ant-design/blob/feature-calendar-hide-type-switch/components/calendar/demo/no-mode-switch.md)
[View rendered components/calendar/index.en-US.md](https://github.com/BenjaVR/ant-design/blob/feature-calendar-hide-type-switch/components/calendar/index.en-US.md)

zombieJ committed:
use ul in list

zombieJ committed:
update snapshot

zombieJ committed:
update comment

Commit:
Added showModeSwitch prop to Calendar to hide the mode switch

Commit:
Added showModeSwitch prop to Calendar to hide the mode switch

Commit:
Added default prop to Calendar Header
+ Updated Jest snapshot

Commit:
Added default prop to Calendar Header
+ Updated Jest snapshot

Commit:
Added docs example

Commit:
Added docs example

Commit:
Renamed some "type" variables to "mode" for consistency

Commit:
Renamed some "type" variables to "mode" for consistency

Commit:
Another Jest snapshot for the markdown doc

Commit:
Another Jest snapshot for the markdown doc

afc163 commented:
You can try rebase feature branch now.

BenjaVR commented:
@afc163 hmm, might have done something wrong?

zombieJ committed:
merge feature branch

zombieJ committed:
use li wrap with comment

yp committed:
Add Italian localization of Icon, Text and global components

Nouzbe committed:
feat: Notification functions accept top, bottom and getContainer as arguments

rinick committed:
fix card mode Tabs border style when used with tabPosition="bottom"

afelicioni committed:
Updates LocaleProvider italian language package

chenshuai2144 committed:
fix #16038, back icon should is a button

chenshuai2144 committed:
snapshots updated

chenshuai2144 committed:
TransButton use div#role=button

Commit:
Extend HTMLAttributes

chenshuai2144 committed:
use operation-unit function

chenshuai2144 committed:
snapshots updated

afc163 committed:
:up: upgrade devDep

chenshuai2144 committed:
Merge branch 'master' into page-header-button

chenshuai2144 committed:
Merge branch 'master' into page-header-button

chenshuai2144 committed:
snapshot updated

afc163 reviewed (changes requested):
Mixed other commits, you should pull and rebase `feature` branch.

zombieJ committed:
adjust hoist-non-react-statics ts import (#16224)

jljsj33 committed:
drawer: add afterVisibleChange

jljsj33 committed:
rm onVisibleChange

zombieJ committed:
fix pagination is null crash (#16231)

zombieJ committed:
update change log (#16233)

zombieJ committed:
fix: Select should trigger onSearch when search input is empty (#16235)

* update rc-select

* update demo

yociduo committed:
fix: Select active item hover color is wrong (#16238)

rinick committed:
Merge remote-tracking branch 'remotes/antd/master'

Commit:
Fix unnecessary indent

zombieJ committed:
fix: Form with forwardRef typescript (#16242)

* update form typescript define with forawardRef

* add ts test

jljsj33 committed:
update

atomoo committed:
remove active style from disabled Switch

afc163 committed:
Merge pull request #16251 from atomoo/fix-switch-active-style

fix: remove active style from disabled Switch

zombieJ committed:
add api collection script & update input-number doc (#16253)

afc163 committed:
Merge pull request #16130 from rinick/master

fix: fix card mode Tabs border style when used with tabPosition="bottom"

afc163 committed:
Merge pull request #16123 from Nouzbe/feature

feat: Make notification functions accept top, bottom and getContainer as arguments

afc163 committed:
Merge pull request #15890 from ant-design/list-ul-li

feat: Wrap List in `ul` for accessibility

afc163 committed:
Merge pull request #16093 from yp/master

Add Italian localization of Icon, Text and global components

afc163 committed:
Merge pull request #16172 from afelicioni/patch-1

Updates LocaleProvider italian language package

afc163 committed:
:white_check_mark: Fix snapshots

afc163 committed:
:up: upgrade jsdom

afc163 committed:
Merge pull request #16228 from ant-design/drawer-afterVisibleChange

feat: Drawer after visible change

afc163 committed:
Merge pull request #16256 from ant-design/fix-snapshot

test: Fix snapshots

afc163 committed:
:memo: upgrade PR template

zombieJ committed:
fix small table header have background color (#16266)

afc163 committed:
feat: 🇭🇷 hr_HR locale (#16258)

* Added Croatian locale

* fixed lint error

* :white_check_mark: Add test cases for hr_HR

* :memo: update i18n documentation

zombieJ committed:
fix: List with actions style should be the right (#16239)

* update css

* should not affect vertical style

* update logic

yoyo837 committed:
modal title break word (#16267)

zombieJ committed:
rise type style priority of Typography (#16275)

zombieJ committed:
docs: add naming rules (#16276)

* add naming rules

* one more space

zombieJ committed:
feat:  add `htmlFor` in Form.Item (#16278)

* add htmlFor in Form.Item

* update doc

zombieJ committed:
add check prepare to make Affix smooth in Safari (#16281)

DiamondYuan committed:
fix: Tree checkbox animation not correct (#16272)

* fix: Tree checkbox animation not correct

* fix: Tree checkbox animation not correct

* fix: Tree checkbox animation not correct

* chore: add comment

zombieJ committed:
merge master

afc163 committed:
Merge pull request #16282 from ant-design/merge-master

Merge master

chenshuai2144 committed:
change onclick dom

chenshuai2144 committed:
beter class name

chenshuai2144 committed:
change use name

zombieJ committed:
feat: Button support `link` type (#16289)

close #15892

zombieJ committed:
fix: Caldendar select option ellipsis when not in `fullscreen` (#16304)

chenshuai2144 committed:
Merge pull request #16187 from ant-design/page-header-button

Back icon should is a button

zombieJ committed:
fix: Pagination active item should have background color

afc163 committed:
Merge pull request #16306 from ant-design/pagination-bg

fix: Pagination active item should have background color

zombieJ committed:
Typography editable mode should support className (#16307)

dependabot-support committed:
Update react-slick requirement from ~0.23.2 to ~0.24.0

Updates the requirements on [react-slick](https://github.com/akiran/react-slick) to permit the latest version.
- [Release notes](https://github.com/akiran/react-slick/releases)
- [Changelog](https://github.com/akiran/react-slick/blob/master/CHANGELOG.md)
- [Commits](https://github.com/akiran/react-slick/compare/0.23.2...0.24.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

zombieJ committed:
chore: Remove useless deps

afc163 committed:
Merge pull request #16332 from ant-design/rm-create-react-class

chore: Remove useless deps

afc163 committed:
Merge pull request #16331 from ant-design/dependabot/npm_and_yarn/react-slick-approx-0.24.0

Update react-slick requirement from ~0.23.2 to ~0.24.0

afc163 committed:
Merge pull request #16195 from swillis12/Fix-tooltip-Type

Fix className prop issue when using Tooltip

afc163 committed:
:memo: changelog for 3.16.6

afc163 committed:
:bug: Add className in TooltipProps only, #16195

afc163 committed:
:memo: English changelog for 3.16.6

afc163 committed:
Merge pull request #16335 from ant-design/changelog-3.16.6

docs: 📝 changelog for 3.16.6

afc163 committed:
:up: upgrade devDependency

afc163 committed:
:lipstick: update code style

sbusch committed:
rc-calendar RangeCalendar calls renderFooter() without args

sbusch committed:
Fix TypeScript def for RangePicker.renderExtraFooter(), gets no args

afc163 committed:
Merge pull request #16341 from sbusch/RangePicker-renderExtraFooter-gets-no-args

RangePicker.renderExtraFooter() gets no args, fix TypeScript definition and clean up code

afc163 committed:
:memo: Improve column.align document

afc163 committed:
Merge pull request #16348 from ant-design/table-doc-improve

docs: Improve column.align document

MrHeer committed:
feat: Add Timeline.Item.position (#16148) (#16193)

* fix: Timeline.pendingDot interface documentation there is a small problem (#16177)

* feat: Add Timeline.Item.position (#16148)

* doc: add version infomation for Timeline.Item.position

chenshuai2144 committed:
fix error types (#16043)

afc163 committed:
:white_check_mark: Fix cascader test snapshot

afc163 committed:
Merge pull request #16356 from ant-design/fix-cascader-snapshot

test: Fix cascader test snapshot

zombieJ committed:
refactor: Update Tree & TreeSelect deps (#16330)

* use CSSMotion

* update snapshot

afc163 committed:
:bug: Fix Drawer wrong animation direction

close #15713
close #15146

zombieJ committed:
update collapse faq demo

zombieJ committed:
Merge remote-tracking branch 'origin/master' into docs

afc163 committed:
Merge pull request #16361 from ant-design/docs

docs: Update collapse FAQ demo

sbusch committed:
Form onFieldsChange(): document third parameter "allFields", remove non-existent fourth parameter "add" from TypeScript definition (#16343)

* there is no fourth parameter, added by mistake in https://github.com/ant-design/ant-design/commit/619b0ed628789ef42dfdb749b8e0933eca2d08ad

* Form onFieldsChange(): document third parameter 'allFields', introduced in https://github.com/ant-design/ant-design/commit/619b0ed628789ef42dfdb749b8e0933eca2d08ad

zombieJ committed:
adjust slider text style (#15128)

* adjust style

* use keep-all instead

zombieJ committed:
Merge pull request #16354 from ant-design/master

Merge master into feature

zombieJ committed:
feat: Collapse support `expandIconPosition` (#16365)

* update doc

* support expandIconPosition

* update snapshot

chenshuai2144 committed:
feat: Breadcrumb  support DropDown (#16315)

* breadcrumbs support drop down menu

* update doc

* add require less

* fix test

* fix md doc

* less code

* fix  style warning

* update snap

* add children render test

zombieJ committed:
feat: TreeNode support checkable

ycjcl868 committed:
fix: Upload types (#16360)

* fix: Upload types

* fix: ci

* fix: ci

* fix: ci

* fix: ci

chenshuai2144 committed:
Merge pull request #16369 from ant-design/treenode-checkable

feat: TreeNode support checkable

zombieJ committed:
fix: raf should keep same id (#16370)

jjalonso committed:
Update index.en-US.md (#16373)

afc163 committed:
Merge pull request #16358 from ant-design/fix-drawer-animation

fix: 🐛 Drawer wrong animation direction

yociduo committed:
feat: add optional to support top and left slick dots (#16186) (#16225)

* add optional to support top and left slick dots

* update carousel snapshot

* Update doc, add placement demo

* update carousel placement demo snapshots

* rename dots placement to position

* update vertical as deprecated

* rename dotsPosition to dotPosition

* refine code

* add warning testcase for vertical

* remove unused warning

* update expression

* Additional test case for dotPosition

afc163 committed:
:zap: Improve upload demo

close #16375

afc163 committed:
:clapper: use new file upload mock API

https://www.mocky.io/

ztplz committed:
Fix some typos

afc163 committed:
Merge pull request #16391 from ztplz/fix_typo

Fix some typos

afc163 committed:
Merge pull request #16390 from ant-design/improve-upload-demo

chore: Improve upload demo

afc163 committed:
:bug: Fix AutoComplete align issue

close #14156

afc163 committed:
Merge pull request #16395 from ant-design/fix-auto-complete-align

fix: 🐛 Fix AutoComplete align issue

zombieJ committed:
fix: Checkbox.Group onChange should not pass removed value (#16392)

* filter removed value

* clean up

* code style update

sbusch committed:
Upgrade TypeScript ECMAScript support from es7 (= ECMAScript 2016) to es2017 (= ECMAScript 2017) (#16340)

0xTars committed:
fix:  username misspelling in the form. (#16398)

* fix normal-login spelling mistake

* fix horizontal-login spelling mistake

* fix index.zh-CN spelling mistake

* fix index.en-US spelling mistake

* update form demo.test.js.snap

sbusch committed:
fix: Add some missing TypeScript prop definititions and clean up date-picker definitions (#16342)

* Add some missing TypeScript prop definitition, clean up date-picker definitions

* use DatePickerMode instead of redundant enumeration of modes

* decade was  missing from the list of modes

See https://codesandbox.io/s/388xplj646
- activate DatePicker
- click on year in header
- click on years in header
- "decade" is printed on footer as mode for renderExtraFooter()

* add decade as possible value for mode prop

Commit:
fix: TypeScript definitions for Select component's onSelect / onDeselect props, update inaccurate docs (#16344)

* Fix value param type for onSelect, onDeselect prop

* Fix docs for Select props (value, defaultValue, onSelect, onDeselect)

zombieJ committed:
fix nest style (#16419)

zombieJ committed:
refactor: Upgrade `rc-tree-select` to support pure React motion (#16402)

* upgrade `rc-tree-select`

* update snapshot

chenshuai2144 committed:
Merge branch 'master' into feature

zombieJ committed:
update doc (#16422)

chenshuai2144 committed:
3.17.0 changelog

chenshuai2144 committed:
fix warning

chenshuai2144 committed:
fix review warning

chenshuai2144 committed:
Merge pull request #16427 from ant-design/changelog

Changelog

chenshuai2144 committed:
meger feature to master (#16421)

* use ul in list

* update snapshot

* update comment

* feat: TreeSelect support `showSearch` in multiple mode (#15933)

* update rc-tree-select

* typo

* update desc & snapshot

* update desc & snapshot

* check default showSearch

* feat: table customizing variable (#15971)

* feat: added table selected row color variable

* fix: @table-selected-row-color default is inherit

* feat: Upload support customize previewFile (#15984)

* support preview file

* use promise

* dealy load

* use canvas of render

* use domHook of test

* update demo

* add snapshot

* update types

* update testcase

* feat: form customizing variables (#15954)

* fix: added styling form input background-color

* feat: added '@form-warning-input-bg' variable

* feat: added '@form-error-input-bg' variable

* use li wrap with comment

* feat: Support append theme less file with less-variable (#16118)

* add override

* add override support

* update doc

* feat: dropdown support set right icon

* docs: update doc of dropdown component

* style: format dropdown-button.md

* test: update updateSnapshot

* style: format dropdown-button.md

* test: update updateSnapshot

* test: update updateSnapshot

* style: change style of dropdown-button demo

* fix: fix document table order

* feat: Support SkeletonAvatarProps.size accept number (#16078) (#16128)

* chore:update style of demo

* feat: Notification functions accept top, bottom and getContainer as arguments

* drawer: add afterVisibleChange

* rm onVisibleChange

* update

* feat: 🇭🇷 hr_HR locale (#16258)

* Added Croatian locale

* fixed lint error

* :white_check_mark: Add test cases for hr_HR

* :memo: update i18n documentation

* feat:  add `htmlFor` in Form.Item (#16278)

* add htmlFor in Form.Item

* update doc

* feat: Button support `link` type (#16289)

close #15892

* feat: Add Timeline.Item.position (#16148) (#16193)

* fix: Timeline.pendingDot interface documentation there is a small problem (#16177)

* feat: Add Timeline.Item.position (#16148)

* doc: add version infomation for Timeline.Item.position

* refactor: Update Tree & TreeSelect deps (#16330)

* use CSSMotion

* update snapshot

* feat: Collapse support `expandIconPosition` (#16365)

* update doc

* support expandIconPosition

* update snapshot

* feat: Breadcrumb  support DropDown (#16315)

* breadcrumbs support drop down menu

* update doc

* add require less

* fix test

* fix md doc

* less code

* fix  style warning

* update snap

* add children render test

* feat: TreeNode support checkable

* feat: add optional to support top and left slick dots (#16186) (#16225)

* add optional to support top and left slick dots

* update carousel snapshot

* Update doc, add placement demo

* update carousel placement demo snapshots

* rename dots placement to position

* update vertical as deprecated

* rename dotsPosition to dotPosition

* refine code

* add warning testcase for vertical

* remove unused warning

* update expression

* Additional test case for dotPosition

* refactor: Upgrade `rc-tree-select` to support pure React motion (#16402)

* upgrade `rc-tree-select`

* update snapshot

* 3.17.0 changelog

* fix warning

* fix review warning

chenshuai2144 committed:
bump 3.17.0

afc163 committed:
Merge pull request #16438 from ant-design/feature

Merge feature into master

afc163 committed:
:bug: Fix hasFeedback overlap issue with Input.Password

close #16443

afc163 committed:
:white_check_mark: update snapshots

afc163 committed:
Merge pull request #16445 from ant-design/fix-password-feedback

fix: Fix password feedback

afc163 committed:
fix filterDropdown type

afc163 committed:
Merge pull request #16446 from ant-design/fix-filterDropdown-ts

fix: improve filterDropdown typescript type

afc163 committed:
:lipstick: Adjust pagination link style

link take all space inside li

close #15704

afc163 committed:
Merge pull request #16447 from ant-design/improve-pagination-style

chore:  💄 Adjust pagination link style

afc163 committed:
:white_check_mark: use prettier cli

afc163 committed:
:lipstick: prittier all files

afc163 committed:
:zap: prettier before commit

afc163 committed:
:white_check_mark: Fix lint

afc163 committed:
:bug: Do not prettier js in less

afc163 committed:
chore: 💄 Prettier all files (Merge pull request #16448 from ant-design/prettier-all)

chore: 💄 Prettier all files

zombieJ committed:
feat: Support Transfer render (#16026)

* clean up

* refactor select all logic

* omit old api to render children

* add search

* remove useless code

* fix logic

* Table demo

* update demo

* fixed min-height

* fix normal style

* style update

* update snapshot

* update snapshot

* update version

* update doc

* update md desc

* update test case

* fix test typo

* update demo

* update snapshot

* typescript correction

* update docs

* add Tree Transfer demo

* default expand all

* update demo

afc163 committed:
:white_check_mark: Resolve conflict with feature branch

afc163 committed:
Merge pull request #16457 from ant-design/resolve-feature-conflict

Resolve feature conflict

zombieJ committed:
update doc (#16459)

afc163 committed:
:white_check_mark: Resolve conflict with feature branch

afc163 committed:
Merge pull request #16460 from ant-design/resolve-feature-conflict-again

Resolve feature conflict again

afc163 committed:
fix: Fix lots of missing icons (#15874)

* :bug: Fix lots of missing icons

close ant-design/ant-design-icons#26
close #15850

* update icons deps

* fix lint

zombieJ committed:
Merge pull request #16453 from ant-design/master

Merge master into feature

afc163 committed:
:bug: Fix disabled Option wrong hover color

afc163 committed:
:memo: optimize site info

afc163 committed:
Merge pull request #16477 from ant-design/fix-disable-option-style

fix: 🐛 Fix Select selected and disabled item wrong hover color

afc163 committed:
:bug: Avoid disabled Upload being triggered by label (#16483)

close #16478

zombieJ committed:
update list ts def (#16485)

afc163 committed:
Update CHANGELOG.zh-CN.md

afc163 committed:
:memo: update changelog

jas0ncn committed:
Fix: Print reject error when ActionButton catch promise rejects

saxenanihal95 committed:
refactor: Render function size reduce (#16244)

* updated after resolving conflicts

* lint error fixes

afc163 committed:
Merge pull request #16437 from jas0ncn/master

fix: Print reject error when ActionButton catch promise rejects

Lyndon001 committed:
Update overview.zh-CN.md

Lyndon001 committed:
Update overview.zh-CN.md

ztplz committed:
Fix hoist-non-react-statics compile error (#16397)

zombieJ committed:
fix: customize icon in notification size not correct (#16507)

nailuoGG committed:
fix doc (#16506)

yociduo committed:
fix(PageHeader): Should not render blank dom

yociduo committed:
Update snapshot for test case

afc163 committed:
Merge pull request #16510 from yociduo/master

fix(PageHeader): Should not render blank dom

zombieJ committed:
update snapshot (#16515)

hansnow committed:
📝 Update Menu docs (#16513)

add missing params in Menu's onClick, onSelect and onDeselect callback

zombieJ committed:
feat: Pagination support `disabled` prop (#16493)

* feat: Pagination support `disable`

* update snapshot

* update style

* update style

* update snapshot

* update font color

ycjcl868 committed:
fix: codesandbox not support slash import (#16509)

* fix: codesandbox not support lodash/*

* fix: remove slash deps

afc163 committed:
Update confirm-promise.md

SylvanasGone committed:
fix[Select]: add missing type definition (#16504)

* fix[Select]: add missing type definition

* update docs

vicrep committed:
fix(forms): improve field error method type defs


Change `getFieldError` and `getFieldsError` to return actual types, instead of very restrictive `Object` type (which should be avoided in general, in favour of `unknown`, `object`, or `Record` depending on actual use).

afc163 committed:
Merge pull request #16524 from vicrep/patch-1

fix(forms): improve field error method type defs

abgaryanharutyun committed:
Just a spelling update

Just a spelling update (fix) for the CSS doc styl to style

afc163 committed:
Merge pull request #16531 from abgaryanharutyun/patch-2

Just a spelling update

Commit:
docs: 📝 changelog for 3.17.1 (#16534)

yutingzhao1991 committed:
merge master

afc163 committed:
Merge pull request #16537 from ant-design/chore/mergemaster

Merge master to feature

zombieJ committed:
update doc (#16545)

Commit:
docs: changelog for 3.18.0 (#16539)

* docs: changelog for 3.18.0

* fix review issue

zombieJ committed:
Merge pull request #16538 from ant-design/feature

Merge feature to master for 3.18.0

zombieJ committed:
update snapshot (#16556)

zombieJ committed:
docs: Add changelog (#16559)

afc163 committed:
Update package.json

afc163 committed:
Merge pull request #16564 from ant-design/afc163-patch-1

Update package.json

zombieJ committed:
Merge pull request #16567 from ant-design/master

chore: feature merge master

zombieJ committed:
feat: Add Mentions component (#16532)

* init

* first demo

* support empty

* add loading support

* add form sample

* update form sample

* omit value & defaultValue

* add 2 rest demo

* placement support

* update docs

* fix test

* update docs

* add test case

* fix lint

* follow textarea style

* update docs style

Commit:
Updated Jest snapshots + fixed merge conflicts
